ledger/participant-integration-api/src/main/scala/platform/configuration/IndexServiceConfig.scala

```diff
@@ -46,3 +58,41 @@ object IndexServiceConfig {
   val DefaultInMemoryFanOutThreadPoolSize: Int = 16
   val PreparePackageMetadataTimeOutWarning: FiniteDuration = FiniteDuration(1, "second")
 }
+
+// TODO etq: Take care what config key names will get exposed in the user-facing participant config
+case class TransactionsFlatStreamReaderConfig(
+    maxIdsPerIdPage: Int = 20000,
+    maxPagesPerIdPagesBuffer: Int = 1,
+    maxWorkingMemoryInBytesForIdPages: Int = 100 * 1024 * 1024,
+    maxPayloadsPerPayloadsPage: Int = 1000, // eventsPageSize
+    maxParallelIdCreateQueries: Int = 4,
+    maxParallelIdConsumingQueries: Int = 4,
+    // TODO etq: This must be a power of 2
+    maxParallelPayloadCreateQueries: Int = 2,
+    // TODO etq: This must be a power of 2
+    maxParallelPayloadConsumingQueries: Int = 2,
+    maxParallelPayloadQueries: Int = 2,
+    payloadProcessingParallelism: Int = 8,
+)
+object TransactionsFlatStreamReaderConfig {
+  def default: TransactionsFlatStreamReaderConfig = TransactionsFlatStreamReaderConfig()
+}
+
+// TODO etq: Take care what config key names will get exposed in the user-facing participant config
+case class TransactionsTreeStreamReaderConfig(
+    maxIdsPerIdPage: Int = 20000,
+    maxPagesPerIdPagesBuffer: Int = 1,
+    maxWorkingMemoryInBytesForIdPages: Int = 100 * 1024 * 1024,
+    maxPayloadsPerPayloadsPage: Int = 1000,
+    maxParallelIdCreateQueries: Int = 8,
```

Comment:
why do we need more oomph for the trees?

Reply:
For flat create events we fetch ids from `pe_id_filter_stakeholder` table,
while for tree create events we fetch from both `pe_id_filter_stakeholder` and `pe_id_filter_non_stakeholder_informee`